unionfs/unionfs.go

```diff
@@ -712,7 +712,12 @@ func (fs *unionFS) GetXAttr(name string, attr string, context *fuse.Context) ([]
 	}
 	r := fs.getBranch(name)
 	if r.branch >= 0 {
-		return fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		value, code := fs.fileSystems[r.branch].GetXAttr(name, attr, context)
+		// We intercept ENOSYS and replace it with ENODATA; returning ENOSYS would cause FUSE to never to GetXAttr() again.
+		if code == fuse.ENOSYS {
+			return nil, fuse.ENODATA
+		}
+		return value, code
```

Comment:
Support xattr is a property of the entire mount point, since FUSE can't decide which parts of the filesystem do or do not support it. Hence the support should also be where the mountpoint is handled.

This is already partly in place, see 7b281480b21a59f54a0c86451e7de13cd9be736f.   - it just needs to transform ENOSYS into ENODATA there.